test/integration/setup/x509pop/gencerts.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	// The "never expires" timestamp from RFC5280. Probably not a good
+	// idea in practice.
+	neverExpires = time.Date(9999, 12, 31, 23, 59, 59, 0, time.UTC)
```

Comment:
unclear why this is required? Can we not set it to time.Now() + intergationTestTimeout ?

Reply:
This script was cribbed from another test where the certificates lived at rest in the repo. I can reduce it to something small since these ones are generated on the fly.